pkg/kudoctl/cmd/diagnostics/collectors.go

```diff
@@ -0,0 +1,132 @@
+package diagnostics
+
+import (
+	"fmt"
+	"io"
+	"path/filepath"
+	"reflect"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
+)
+
+// Ensure collector is implemented
+var _ collector = &resourceCollector{}
+
+// resourceCollector - collector interface implementation for Kubernetes resources (runtime objects)
+type resourceCollector struct {
+	loadResourceFn func() (runtime.Object, error)
+	name           string               // object kind used to describe the error
+	parentDir      stringGetter         // parent dir to attach the printer's output
+	failOnError    bool                 // define whether the collector should return the error
```

Comment:
Do we need `failOnError` parameter? The only case when collector that should fail on error is when collecting an `Instance` resource. Can we just hardcode it?